openstack/tools/func_test_tools/common.sh

```diff
@@ -44,3 +44,30 @@ apply_func_test_pr ()
         -f './src/test-requirements*.txt' \
         "$msg"
 }
+
+allocate_port ()
+{
+    local net_name=$1
+    local port_name=$2
+    openstack port create --network $net_name $port_name
+}
+
+create_zaza_vip ()
+{
+    """
+    Allocates a vip ensuring to use existing ones if they exist.
+
+    Returns the address of the vip.
+    """
+    local vip_id=$1
+    # We use the same naming convention as ../tools/allocate_vips.sh to avoid conflicts and re-use
+    # those vips.
+    vip_port_name=ps6-vip-ip$((vip_id-1))
+    vip_addr=$(openstack port show -c fixed_ips $vip_port_name -f yaml| yq .fixed_ips[0].ip_address)
+    if [[ $vip_addr = null ]]; then
+        # Pre-allocate ports with addresses used for VIPs so that they don't
+        # collide with the deployment itself.
+        allocate_port net_${OS_USERNAME}-psd $vip_port_name
+    fi
+    echo $vip_addr
```

Comment:
vip_addr will not be valid here if we created a port - I think you can get the value by using -c -f with 'port create' but haven't tried it.

Reply:
thank you, will fix